src/main/java/net/snowflake/ingest/streaming/internal/ConfigureCallHandler.java

```
@@ -0,0 +1,188 @@
+
+    public ConfigureCallHandlerBuilder setSchema(String schema) {
+      this.schema = schema;
+      return this;
+    }
+
+    public ConfigureCallHandlerBuilder setTable(String table) {
+      this.table = table;
+      return this;
+    }
+
+    public ConfigureCallHandlerBuilder setIsTestMode(boolean isTestMode) {
+      this.isTestMode = isTestMode;
+      return this;
+    }
+
+    public ConfigureCallHandler build() {
+      return new ConfigureCallHandler(this);
+    }
+  }
+
+  ConfigureCallHandler(ConfigureCallHandlerBuilder builder) {
+    if (!builder.isTestMode) {
+      Utils.assertNotNull("http client", builder.httpClient);
+      Utils.assertNotNull("request builder", builder.requestBuilder);
+      Utils.assertNotNull("api name", builder.ApiName);
+      Utils.assertStringNotNullOrEmpty("configure endpoint", builder.configureEndpoint);
+    }
+
+    this.httpClient = builder.httpClient;
+    this.requestBuilder = builder.requestBuilder;
+    this.apiName = builder.ApiName;
+    this.configureEndpoint = builder.configureEndpoint;
+    this.role = builder.role;
+    this.database = builder.database;
+    this.schema = builder.schema;
+    this.table = builder.table;
+  }
+
+  /**
+   * Make a configure call to the Snowflake service
+   *
+   * @return the configure response
+   * @throws IOException
+   */
+  ConfigureResponse makeConfigureCall() throws IOException {
+    return makeConfigureCall(makeConfigurePayload());
+  }
+
+  /**
+   * Make a configure call to the Snowflake service with a file name, used for GCS
+   *
+   * @param fileName the file name
+   * @return the configure response
+   * @throws IOException
+   */
+  ConfigureResponse makeConfigureCall(String fileName) throws IOException {
+    Map<String, String> payload = makeConfigurePayload();
+    payload.put("file_name", fileName);
+    return makeConfigureCall(payload);
+  }
+
+  private ConfigureResponse makeConfigureCall(Map<String, String> payload) throws IOException {
```

its really this one API call thats being wrapped by this call. Suggest the following:
- Introduce a SnowflakeServiceClient.java class that takes a CloseableHttpClient as a ctor argument
- Have one method per server API in this class, with the method's signature taking in a strongly typed object (ConfigureClientRequest in this case) as the method argument
- Respond with strongly-typed response objects (ConfigureClientResponse)
- hide all error handling and request serialization and response serialization inside this one class

Since each API call is pretty simple we don't need a separate class/handler object for each API call.

Eventually we should only be passing around this SnowflakeServiceClient.java object around instead of the current practice of passing around CloseableHttpClient.

---

Created `SnowflakeServiceClent`